base/libgit2/types.jl

```diff
@@ -612,13 +631,54 @@ function getobjecttype(obj_type::Cint)
     end
 end
 
-type UserPasswordCredentials <: AbstractPayload
+"Empty credentials"
+type EmptyCredentials <: AbstractCredentials end
+
+"Credentials that support only `user` and `password` parameters"
+type UserPasswordCredentials <: AbstractCredentials
     user::AbstractString
     pass::AbstractString
-    used::Bool
-    UserPasswordCredentials(u::AbstractString,p::AbstractString) = new(u,p,false)
+    count::Int
+    use_ssh_agent::UInt8
+    UserPasswordCredentials(u::AbstractString,p::AbstractString) = new(u,p,3,zero(UInt8))
+end
+function isused(p::UserPasswordCredentials)
+    p.count <= 0 && return true
+    p.count -= 1
```

Comment:
what is this counting, and why does `isused` decrement it? it should be called `isused` if it's mutating